### PR TITLE
fix: artifact subdir error when using volumeMount

### PR DESCRIPTION
**Pull request by tczhao**
Fixes https://github.com/argoproj/argo-workflows/issues/12174

I tried contacting original author from https://github.com/argoproj/argo-workflows/pull/12200 but met no success

We want to expedite this fix and have this ready for the next patch release

### Motivation

To make sure if an artifact contains a subdir and is mounted using volume claim, the workflow runs without failing.
VolumeMount is necessary if the user wants to use ephemeral storage.

### Modifications

Make sure the executor creates a subdirectory before downloading files

### Verification

Added new e2e test -> e2e failed (verified original issue)
Added code change -> e2e succeed (verified fixed the issue)

### CODE REVIEW

**Review by terrytangyuan — APPROVED** [review]
Welcome back!

**Comment by juliev0** [comment]
Hey @tczhao - I'm not sure how much you plan on contributing going forward but in case you're interested, I wanted to link you to [this](https://github.com/argoproj/argo-workflows/blob/main/community/sustainability_effort.md) just in case you haven't seen it :)

**Comment by tczhao** [comment]
> Hey @tczhao - I'm not sure how much you plan on contributing going forward but in case you're interested, I wanted to link you to [this](https://github.com/argoproj/argo-workflows/blob/main/community/sustainability_effort.md) just in case you haven't seen it :)

Hi @juliev0, thanks for letting me know.
I should sign up for that.

File: workflow/executor/executor.go [hunk 1]
```diff
@@ -204,6 +204,11 @@ func (we *WorkflowExecutor) LoadArtifacts(ctx context.Context) error {
 		// the file is a tarball or not. If it is, it is first extracted then renamed to
 		// the desired location. If not, it is simply renamed to the location.
 		tempArtPath := artPath + ".tmp"
+		// Ensure parent directory exist, create if missing
+		tempArtDir := filepath.Dir(tempArtPath)
+		if err := os.MkdirAll(tempArtDir, 0o700); err != nil {
```
Comment (bishengliu7) [inline workflow/executor/executor.go:209]: question: why do we set this 700?

Comment (tczhao) [reply workflow/executor/executor.go:209]: The code here creates a temporary path, then artifact driver will load(download) the object to this path.
The minimum dir permission across the artifact drivers is 700.
Are you experiencing an issue with this?

**Pull request merged**